src/main/java/slimeknights/tconstruct/plugin/crt/managers/TinkerStationManager.java

```diff
@@ -0,0 +1,99 @@
+import slimeknights.tconstruct.library.modifiers.Modifier;
+import slimeknights.tconstruct.library.modifiers.ModifierEntry;
+import slimeknights.tconstruct.library.recipe.RecipeTypes;
+import slimeknights.tconstruct.library.recipe.tinkerstation.ITinkerStationRecipe;
+import slimeknights.tconstruct.library.recipe.tinkerstation.modifier.IncrementalModifierRecipe;
+import slimeknights.tconstruct.library.recipe.tinkerstation.modifier.ModifierMatch;
+import slimeknights.tconstruct.library.recipe.tinkerstation.modifier.ModifierRecipe;
+import slimeknights.tconstruct.library.recipe.tinkerstation.modifier.OverslimeModifierRecipe;
+import slimeknights.tconstruct.plugin.crt.CRTHelper;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@ZenRegister
+@ZenCodeType.Name("mods.tconstruct.TinkerStation")
+public class TinkerStationManager implements IRecipeManager {
+  
+  @ZenCodeType.Method
+  public void addOverslimeModifierRecipe(String name, IIngredient ingredient, int restoreAmount) {
+    name = fixRecipeName(name);
+    ResourceLocation id = new ResourceLocation("crafttweaker", name);
+    OverslimeModifierRecipe recipe = new OverslimeModifierRecipe(id, ingredient.asVanillaIngredient(), restoreAmount);
+    CraftTweakerAPI.apply(new ActionAddRecipe(this, recipe));
+  }
+  
+  @ZenCodeType.Method
+  public void addModifierRecipe(String name, IIngredientWithAmount[] inputs, IIngredient toolRequired, String modifierResult, int modifierResultLevel, int maxLevel, int upgradeSlots, int abilitySlots) {
+    name = fixRecipeName(name);
+    Modifier resultModifier = CRTHelper.getModifier(modifierResult);
+    ResourceLocation id = new ResourceLocation("crafttweaker", name);
+    List<SizedIngredient> collect = Arrays.stream(inputs).map(iIngredientWithAmount -> SizedIngredient.of(iIngredientWithAmount.getIngredient().asVanillaIngredient(), iIngredientWithAmount.getAmount())).collect(Collectors.toList());
+    ModifierMatch entry = ModifierMatch.ALWAYS;
+    ModifierEntry result = new ModifierEntry(resultModifier, modifierResultLevel);
+    ModifierRecipe recipe = new ModifierRecipe(id, collect, toolRequired.asVanillaIngredient(), entry, "", result, maxLevel, upgradeSlots, abilitySlots);
+    CraftTweakerAPI.apply(new ActionAddRecipe(this, recipe));
+  }
+  
+  @ZenCodeType.Method
+  public void addModifierRecipe(String name, IIngredientWithAmount[] inputs, IIngredient toolRequired, String modifierResult, int modifierResultLevel, int maxLevel, int upgradeSlots, int abilitySlots, IData modifierRequirements, int minMatch, String requirementsError) {
+    name = fixRecipeName(name);
+    Modifier resultModifier = CRTHelper.getModifier(modifierResult);
+    ModifierMatch[] modifierMatches = modifierRequirements.asMap().entrySet().stream().map(stringIDataEntry -> ModifierMatch.entry(CRTHelper.getModifier(stringIDataEntry.getKey()), stringIDataEntry.getValue().asNumber().getInt())).toArray(ModifierMatch[]::new);
```

Comment:
ModifierMatch is a recursive data structure, so its a bit weird to see a non-recursive method parsing the data. If you don't want to support it thats fine, just making sure I understand how you implemented this.